### PR TITLE
Worker Xp adjustments

### LINKED ISSUE #5987: Library students not gaining intelligence
### Prerequisites 

- [ ] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.15.2
- MineColonies Version: MineColonies Official pack v3.0
- Structurize Version: MineColonies Official pack v3.0
- Related Mods and their Versions: none

## Expected behavior
Library students would gain intelligence from working at the library

## Actual behavior
I left students in there for several in-game weeks, and they haven't gained any levels. Their home is level 5 and the library is level 4. I gave them several stacks of paper.

## Steps to reproduce the issue

1. Assign colonists to a library
2. Wait
3. 

### Logs

- latest.log: This is on the 1.15 patreon server, unless you need client logs I can't help
- crashlog: 

### Notes
The 3rd page of their GUI, where normally the skills affected by their job are, is blank.

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by someaddons**
Closes #5987

# Changes proposed in this pull request:
This adjusts all workers base XP gain to be roughly on the same level. Some workers got an XP buff some got nerfs.
The overall base XP gain should be around 10xp/day, which is before any xp increasing modifiers. See https://github.com/ldtteam/Dev-Wiki/issues/7 for details on the base.

Fix Herder trying to breed adult animals incorrectly
Fix custom arrows despawning too fast for the archer trainees to evaluate them
Fix knight trainee's fighting with previously assigned citizen

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/CustomArrowEntity.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
     /**
      * Max time the arrow is stuck before removing it
      */
-    private static final int MAX_TIME_IN_GROUND = 10;
+    private static final int MAX_TIME_IN_GROUND = 100;
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/CustomArrowEntity.java:21]: How about we init an arrow with a custom time, so for the arrows of our trainees we can make this 100, and all other stays 10?

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/herders/AbstractEntityAIHerder.java [hunk 2]
```diff
@@ -204,6 +208,17 @@ else if (canFeedChildren() && numOfFeedableAnimals > 0)
         return START_WORKING;
     }
 
+    /**
+     * Checks if we can breed this entity
+     *
+     * @param entity
+     * @return
+     */
+    protected boolean isBreedAble(final AnimalEntity entity)
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/herders/AbstractEntityAIHerder.java:217]: the uppercase "A" is weird =D

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/herders/AbstractEntityAIHerder.java:217]: and @return true if breedable.

File: src/main/java/com/minecolonies/coremod/entity/CustomArrowEntity.java (hunk 1 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/entity/CustomArrowEntity.java:21]: hm dont think we need that, those 5 sec should be still fine for all needs

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/herders/AbstractEntityAIHerder.java (hunk 2 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/herders/AbstractEntityAIHerder.java:217]: intellij complains with small A though so I guess its correct xD

File: src/main/java/com/minecolonies/coremod/entity/CustomArrowEntity.java (hunk 1 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/CustomArrowEntity.java:21]: Oh the motivation is more to have it remove them quicker if we don't need them 5s. (Even less lag)

Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/entity/CustomArrowEntity.java:21]: ye doesnt make a difference really, both despawn rather quick

**Review by Mekle001 — APPROVED** [review]

**Pull request merged**